src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -71,11 +71,18 @@ public static class WebScriptHostBuilderExtension
                 {
                     var webHostEnvironment = rootServiceProvider.GetService<IScriptWebHostEnvironment>();
                     var environment = rootServiceProvider.GetService<IEnvironment>();
+                    var shutdownManager = rootServiceProvider.GetService<IApplicationLifetimeManager>();
+
                     if (FunctionsSyncManager.IsSyncTriggersEnvironment(webHostEnvironment, environment))
                     {
                         services.AddSingleton<IHostedService, FunctionsSyncService>();
                     }
 
+                    if (shutdownManager == null)
+                    {
+                        services.AddSingleton<IApplicationLifetimeManager, ApplicationLifetimeManager>();
```

Comment:
What is this for? haven't you already added the service earlier in AddWebJobsScriptHost?